include/z64save.h

```diff
@@ -59,9 +59,42 @@ typedef enum {
 #define MAGIC_NORMAL_METER 0x30
 #define MAGIC_DOUBLE_METER (2 * MAGIC_NORMAL_METER)
 
+typedef enum {
+    /* 0 */ IBTN_BC_B,
+    /* 1 */ IBTN_BC_C_LEFT,
+    /* 2 */ IBTN_BC_C_DOWN,
+    /* 3 */ IBTN_BC_C_RIGHT,
+    /* 4 */ IBTN_BC_MAX
+} InteractButtonBC;
+
+#define IBTN_BC_C_FIRST IBTN_BC_C_LEFT
+#define IBTN_BC_C_LAST IBTN_BC_C_RIGHT
+
+typedef enum {
+    /* 0 */ IBTN_C_C_LEFT,
+    /* 1 */ IBTN_C_C_DOWN,
+    /* 2 */ IBTN_C_C_RIGHT,
+    /* 3 */ IBTN_C_MAX
+} InteractButtonC;
+
+#define IBTN_C_TO_BC(btnsC) ((btnsC) + 1)
+#define IBTN_BC_TO_C(btnsBC) ((btnsBC) - 1)
+
+typedef enum {
+    /* 0 */ IBTN_BCA_B,
+    /* 1 */ IBTN_BCA_C_LEFT,
+    /* 2 */ IBTN_BCA_C_DOWN,
+    /* 3 */ IBTN_BCA_C_RIGHT,
+    /* 4 */ IBTN_BCA_A,
+    /* 5 */ IBTN_BCA_MAX
+} InteractButtonBCA;
+
+#define IBTN_C_TO_BCA(btnsC) ((btnsC) + 1)
+#define IBTN_BC_TO_BCA(btnsBC) (btnsBC)
```

Comment:
I think it's too many abbreviations compact in a small place that throws me off for this. I'd have no issues fully spelling out `INTERACT_BTN`. What about:

```c
typedef enum {
    /* 0 */ INTERACT_BC_BTN_B,
    /* 1 */ INTERACT_BC_BTN_C_LEFT,
    /* 2 */ INTERACT_BC_BTN_C_DOWN,
    /* 3 */ INTERACT_BC_BTN_C_RIGHT,
    /* 4 */ INTERACT_BC_BTN_MAX
} InteractBCButton;

#define INTERACT_BC_BTN_C_FIRST INTERACT_BC_BTN_C_LEFT
#define INTERACT_BC_BTN_C_LAST INTERACT_BC_BTN_C_RIGHT

typedef enum {
    /* 0 */ INTERACT_C_BTN_C_LEFT,
    /* 1 */ INTERACT_C_BTN_C_DOWN,
    /* 2 */ INTERACT_C_BTN_C_RIGHT,
    /* 3 */ INTERACT_C_BTN_MAX
} InteractCButton;

#define INTERACT_C_BTN_TO_BC_BTN(btnsC) ((btnsC) + 1)
#define INTERACT_BC_BTN_TO_C_BTN(btnsBC) ((btnsBC) - 1)

typedef enum {
    /* 0 */ INTERACT_BCA_BTN_B,
    /* 1 */ INTERACT_BCA_BTN_C_LEFT,
    /* 2 */ INTERACT_BCA_BTN_C_DOWN,
    /* 3 */ INTERACT_BCA_BTN_C_RIGHT,
    /* 4 */ INTERACT_BCA_BTN_A,
    /* 5 */ INTERACT_BCA_BTN_MAX
} InteractBCAButton;

#define INTERACT_C_BTN_TO_BCA_BTN(btnsC) ((btnsC) + 1)
#define INTERACT_BC_BTN_TO_BCA_BTN(btnsBC) (btnsBC)
```

Reply:
I like this yeah, and I think it works with "interact"

the thing with "interact" is "there are other buttons you use to 'interact'", but this way e.g. `INTERACT_C_BTN_` could read as "the interact C button(s)"

Reply:
I went ahead with this at least for now